pallets/pips/src/benchmarking.rs

```diff
@@ -0,0 +1,450 @@
+            proposal,
+            42.into(),
+            Some(url),
+            Some(description)
+        )?;
+        let origin = T::VotingMajorityOrigin::successful_origin();
+        Module::<T>::set_prune_historical_pips(RawOrigin::Root.into(), false)?;
+        let reject_call = Call::<T>::reject_proposal(0);
+        reject_call.dispatch_bypass_filter(origin.clone())?;
+        let call = Call::<T>::prune_proposal(0);
+    }: {
+        call.dispatch_bypass_filter(origin)?;
+    }
+    verify {
+        assert_eq!(false, Proposals::<T>::contains_key(&0));
+        assert_eq!(false, ProposalMetadata::<T>::contains_key(&0));
+    }
+
+    reschedule_execution {
+        // length of the proposal padding
+        let c in 0 .. 100_000;
+
+        let (proposer_account, proposer_origin, proposer_did) = make_account::<T>("proposer", 0);
+        identity::CurrentDid::put(proposer_did);
+        let (proposal, url, description) = make_proposal::<T>(c as usize);
+        Module::<T>::set_proposal_cool_off_period(RawOrigin::Root.into(), 0.into())?;
+        Module::<T>::propose(
+            proposer_origin.clone().into(),
+            proposal,
+            42.into(),
+            Some(url.clone()),
+            Some(description.clone())
+        )?;
+        T::GovernanceCommittee::bench_set_release_coordinator(proposer_did);
+        Module::<T>::snapshot(proposer_origin.clone().into())?;
+        let enact_origin = T::VotingMajorityOrigin::successful_origin();
+        let enact_call = Call::<T>::enact_snapshot_results(vec![(0, SnapshotResult::Approve)]);
+        enact_call.dispatch_bypass_filter(enact_origin)?;
+        let future_block = frame_system::Module::<T>::block_number() + 100.into();
+    }: _(proposer_origin, 0, Some(future_block))
+    verify {
+        assert_eq!(true, PipToSchedule::<T>::contains_key(&0));
+    }
+
+    clear_snapshot {
+        // length of the proposal padding
+        let c in 0 .. 100_000;
+
+        let (account, origin, did) = make_account::<T>("proposer", 0);
+        identity::CurrentDid::put(did);
+        let (proposal, url, description) = make_proposal::<T>(c as usize);
+        Module::<T>::set_proposal_cool_off_period(RawOrigin::Root.into(), 0.into())?;
+        Module::<T>::propose(
+            origin.clone().into(),
+            proposal,
+            42.into(),
+            Some(url.clone()),
+            Some(description.clone())
+        )?;
+        T::GovernanceCommittee::bench_set_release_coordinator(did);
+        Module::<T>::snapshot(origin.clone().into())?;
+        assert!(SnapshotMeta::<T>::get().is_some());
+    }: _(origin)
+    verify {
+        assert!(SnapshotMeta::<T>::get().is_none());
+    }
+
+    snapshot {
+        // length of the proposal padding
+        let c in 0 .. 100_000;
+        // number of proposals
+        let p in 0 .. 5;
+        // first group of voters
+        let h in 0 .. 10;
+        // second group of voters
+        let b in 0 .. 10;
```

Comment:
We need much bigger numbers for the tests. Depending on the result, we can restrict these in our runtime.

Every number/variable you are restricting here should also be restricted in the runtime.

Reply:
With the current values the benchmark run time is longer than 10 minutes on an EPYC. The observed benchmark setup time complexity appears to be superlinear in `p`, `h` and `b` (there is an odd feature that amounts to disregarding either `h` or `b` in some cases). Therefore increasing the upper bounds will lead to longer run times. Do we really need this given that `h` and `b` are setup parameters with no particular relevance to the extrinsics?

`h` and `b` are intended to simulate distributions of aye/nay votes across the queued proposals. However there is no direct relationship with actual numbers of votes. The only way to provide the actual numbers to the benchmark is to give it the queued proposals as input. Currently I don't see any other way than doing this by adjusting the weight in post-dispatch.

Reply:
For local execution, just reduce the steps to a small number. That'll reduce the overall benchmark time. I'll run the final benchmarks on a dedicated server. I might just let it run for a week to get reliable results :).

FWIW, we only need to increase `p`. `h` and `b`, as you say, are irrelevant and we can remove them from this benchmark. We can just case 1 vote per proposal and get the same overall complexity.